File: src/main/java/ru/arc/hooks/zauction/AuctionMessager.java
package ru.arc.hooks.zauction;

import java.util.List;

import lombok.RequiredArgsConstructor;
import ru.arc.network.ChannelListener;
import ru.arc.network.RedisManager;
import ru.arc.util.Common;

import static ru.arc.util.Logging.error;

@RequiredArgsConstructor
public class AuctionMessager implements ChannelListener {
    public final String channel, channelAll;
    private final RedisManager redisManager;

    @Override
    public void consume(String channel, String message, String originServer) {

    }

    public void send(List<AuctionItemDto> itemDtoList) {
        try {
            redisManager.publish(channel, Common.gson.toJson(itemDtoList));
        } catch (Exception e) {
            error("Error sending auction items", e);
        }
    }
}

File: src/main/kotlin/ru/arc/hooks/zauction/AuctionMessager.kt
package ru.arc.hooks.zauction

import ru.arc.redis.ChannelListener
import ru.arc.redis.RedisOperations
import ru.arc.util.Common
import ru.arc.util.Logging.error

class AuctionMessager(
    val channel: String,
    val channelAll: String,
    val saleChannel: String,
    private val redis: RedisOperations,
) : ChannelListener {

    override fun consume(channel: String, message: String, originServer: String) {}

    fun send(itemDtoList: List<AuctionItemDto>) {
        try {
            // Every scheduled broadcast is the complete shared-auction snapshot.
            // Publishing it as a partial update would leave sold, returned, or
            // expired listings stuck in ProxyARC's in-memory Discord feed.
            redis.publish(channelAll, Common.gson.toJson(itemDtoList))
        } catch (e: Exception) {
            error("Error sending auction items", e)
        }
    }

    fun sendSale(event: AuctionSaleEventDto) {
        try {
            redis.publish(saleChannel, Common.gson.toJson(event))
        } catch (e: Exception) {
            error("Error sending auction sale event", e)
        }
    }
}
